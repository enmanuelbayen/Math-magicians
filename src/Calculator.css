body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family:
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    'Roboto',
    'Oxygen',
    'Ubuntu',
    'Cantarell',
    'Fira Sans',
    'Droid Sans',
    'Helvetica Neue',
    sans-serif;
  font-size: 1.2rem;
  font-weight: 600;
  line-height: 1.5;
}

img,
video,
svg {
  max-width: 100%;
  display: block;
}

.orange {
  background-color: #ff5900d2;
}

.App {
  display: flex;
  justify-content: center;
}

.calculator {
  padding: 10% 0;
}

.calculator button {
  cursor: pointer;
}

.result {
  text-align: end;
  background: #a4a1a0;
  padding: 0.7em;
  color: azure;
  font-size: 1.4em;
}

.grid {
  display: grid;
}

.key-container {
  grid-template-columns: repeat(4, 6em);
}

.key-row {
  width: 100%;
  padding: 2em;
  border: solid 0.5px #a4a1a0;
}

#ceroKey {
  grid-column: span 2;
}
