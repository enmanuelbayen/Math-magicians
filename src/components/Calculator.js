import React, { useState } from 'react';
import Keys from './Keys';
import calculate from '../Logic/calculate';

const Calculator = () => {
  const [operator, setOperator] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setOperator(calculate(operator, buttonName));
  };

  const output = operator.next ? operator.next : operator.total || 0;

  return (
    <div className="calculator">
      <div className="result">{output}</div>
      <Keys onClick={handleClick} />
    </div>
  );
};

export default Calculator;
