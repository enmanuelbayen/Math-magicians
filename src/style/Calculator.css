.orange {
  background-color: #ff5900d2;
}

.App {
  margin: 0;
  grid-template-columns: repeat(2, 1fr);  
}

.calculator {
 padding: 2%;
}

.calculator button {
  cursor: pointer;
}

.result {
  text-align: end;
  background: #a4a1a0;
  padding: 0.7em;
  color: azure;
  font-size: 1.4em;
}

.grid {
  display: grid;
}

.key-container {
  grid-template-columns: repeat(4, 1fr);
}

.key-row {
  cursor: pointer;
  padding: 2em 1em;
  border: solid 0.5px #a4a1a0;
}

.key-row:hover {
  background: #4864c1;
}

#ceroKey {
  grid-column: span 2;
}
