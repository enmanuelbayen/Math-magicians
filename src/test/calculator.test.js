import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  test('Renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });

  test('Handles button clicks correctly', () => {
    const { getByText } = render(<Calculator />);
    const button = getByText('7');

    fireEvent.click(button);

    expect(button).toMatchSnapshot();
  });
});
