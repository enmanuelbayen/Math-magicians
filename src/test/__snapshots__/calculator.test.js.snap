// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Handles button clicks correctly 1`] = `
<button
  class="key-row"
  type="button"
>
  7
</button>
`;

exports[`Calculator component Renders correctly 1`] = `
<div>
  <div
    class="calculator"
  >
    <div
      class="result"
    >
      0
    </div>
    <div
      class="key-container grid"
    >
      <button
        class="key-row"
        type="button"
      >
        AC
      </button>
      <button
        class="key-row"
        type="button"
      >
        +/-
      </button>
      <button
        class="key-row"
        type="button"
      >
        %
      </button>
      <button
        class="key-row orange"
        type="button"
      >
        รท
      </button>
      <button
        class="key-row"
        type="button"
      >
        7
      </button>
      <button
        class="key-row"
        type="button"
      >
        8
      </button>
      <button
        class="key-row"
        type="button"
      >
        9
      </button>
      <button
        class="key-row orange"
        type="button"
      >
        x
      </button>
      <button
        class="key-row"
        type="button"
      >
        4
      </button>
      <button
        class="key-row"
        type="button"
      >
        5
      </button>
      <button
        class="key-row"
        type="button"
      >
        6
      </button>
      <button
        class="key-row orange"
        type="button"
      >
        -
      </button>
      <button
        class="key-row"
        type="button"
      >
        1
      </button>
      <button
        class="key-row"
        type="button"
      >
        2
      </button>
      <button
        class="key-row"
        type="button"
      >
        3
      </button>
      <button
        class="key-row orange"
        type="button"
      >
        +
      </button>
      <button
        class="key-row"
        id="ceroKey"
        type="button"
      >
        0
      </button>
      <button
        class="key-row"
        type="button"
      >
        .
      </button>
      <button
        class="key-row orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
